db/background_mgr_attachment_compaction.go

```diff
@@ -142,9 +191,30 @@ func (a *AttachmentCompactionManager) Run(ctx context.Context, options map[strin
 		fallthrough
 	case "cleanup":
 		a.SetPhase("cleanup")
-		persistClusterStatus()
-		err := attachmentCompactCleanupPhase(ctx, dataStore, collectionID, database, a.CompactID, a.VBUUIDs, terminator)
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			persistClusterStatus()
+			err = attachmentCompactCleanupPhase(ctx, dataStore, collectionID, database, a.CompactID, a.VBUUIDs, terminator)
+			if errors.As(err, &rollbackErr) || errors.Is(err, base.ErrVbUUIDMismatch) {
```

Comment:
why does this check 

```
if errors.As(err, &rollbackErr) || errors.Is(err, base.ErrVbUUIDMismatch)
```

in cleanup but not in mark or sweep?

Reply:
Added it to the mark phase. We don't have rollback handling on the sweep phase, if rollback is hit here user needs to manually reset the process.